aws/table_aws_route53_resolver_endpoint.go

```diff
@@ -0,0 +1,315 @@
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.GetResolverEndpointInput{
+		ResolverEndpointId: &id,
+	}
+
+	// Get call
+	data, err := svc.GetResolverEndpoint(params)
+	if err != nil {
+		logger.Debug("getAwsRoute53ResolverEndpoint", "ERROR", err)
+		return nil, err
+	}
+
+	return data.ResolverEndpoint, nil
+}
+
+func listResolverEndpointIPAddresses(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listResolverEndpointIpAddresses")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	resolverEndpointData := h.Item.(*route53resolver.ResolverEndpoint)
+
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.ListResolverEndpointIpAddressesInput{
+		ResolverEndpointId: resolverEndpointData.Id,
+	}
+
+	op, err := svc.ListResolverEndpointIpAddresses(params)
+	if err != nil {
+		logger.Debug("getMaintenanceWindowTargets", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func listResolverRule(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listResolverRule")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	resolverEndpointData := h.Item.(*route53resolver.ResolverEndpoint)
+	id := resolverEndpointData.Id
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.ListResolverRulesInput{
+		Filters: []*route53resolver.Filter{
+			{
+				Name: types.String("ResolverEndpointId"),
+				Values: []*string{
+					id,
+				},
+			},
+		},
+	}
+
+	op, err := svc.ListResolverRules(params)
+	if err != nil {
+		logger.Debug("listResolverRule", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func getAwsRoute53ResolverEndpointTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsRoute53ResolverEndpointTags")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	maintenanceWindowData := h.Item.(*route53resolver.ResolverEndpoint)
```

Comment:
Please use appropriate variable instead of `maintenanceWindowData` and change accordingly.